server/portal/apps/workspace/api/views.py

```diff
@@ -44,16 +45,27 @@ def _get_user_app_license(license_type, user):
     return lic
 
 
+def _get_all_exec_systems(user):
+    """List of all enabled execution systems available for the user."""
+    tapis = user.tapis_oauth.client
+    return tapis.systems.getSystems(listType="ALL", select="allAttributes", search="(canExec.eq.true)~(enabled.eq.true)")
+
+
 def _get_app(app_id, app_version, user):
     tapis = user.tapis_oauth.client
     if app_version:
         app_def = tapis.apps.getApp(appId=app_id, appVersion=app_version)
     else:
         app_def = tapis.apps.getAppLatestVersion(appId=app_id)
+
     data = {'definition': app_def}
 
-    # GET EXECUTION SYSTEM INFO TO PROCESS SPECIFIC SYSTEM DATA E.G. QUEUE INFORMATION
-    data['exec_sys'] = tapis.systems.getSystem(systemId=app_def.jobAttributes.execSystemId)
+    has_dynamic_exec_system = getattr(app_def.notes, 'dynamicExecSystem', False)
```

Comment:
Should we add one more field to `notes` called `dynamicExecSystems`, which we could specify if we wanted to further filter the systems we wanted the app to be available on?

Use case scenario: we know a modularized app works on `frontera` and `ls6`, but not `stampede3` or even a VM, so don't want it to show there.

Reply:
Yes, made those changes. Was fixing tests, this PR is updated.